doc/examples/bundle_extraction.py

```diff
@@ -0,0 +1,165 @@
+"""
+==================================================
+Automatic Fiber Bundle Extraction with RecoBundles
+==================================================
+
+This example explains how we can use RecokBundles [Garyfallidis17]_ to
+extract bundles from tractograms.
+
+First import the necessary modules.
+"""
+
+from dipy.segment.bundles import RecoBundles
+from dipy.align.streamlinear import whole_brain_slr
+from dipy.viz import window, actor
+from dipy.io.streamline import load_trk, save_trk
+
+
+"""
+Download and read data for this tutorial
+"""
+
+from dipy.data.fetcher import (fetch_target_tractogram_hcp,
+                               fetch_bundle_atlas_hcp842,
+                               read_bundle_atlas_hcp842,
+                               read_target_tractogram_hcp)
+
+fetch_target_tractogram_hcp()
+fetch_bundle_atlas_hcp842()
+atlas_file, all_bundles_files = read_bundle_atlas_hcp842()
+target_file = read_target_tractogram_hcp()
+
+atlas, atlas_header = load_trk(atlas_file)
+target, target_header = load_trk(target_file)
```

Comment:
As explained before, `read_bundle_atlas_hcp842` and `read_target_tractogram_hcp` should return the Streamlines object and you get all filenames by the fetcher